src/uct/ib/ud/base/ud_ep.c

```diff
@@ -493,26 +493,33 @@ static ucs_status_t uct_ud_ep_disconnect_from_iface(uct_ep_h tl_ep)
     return UCS_OK;
 }
 
-ucs_status_t uct_ud_ep_create_connected_common(uct_ud_iface_t *iface,
-                                               const uct_ib_address_t *ib_addr,
-                                               const uct_ud_iface_addr_t *if_addr,
-                                               unsigned path_index,
-                                               uct_ud_ep_t **new_ep_p,
-                                               uct_ud_send_skb_t **skb_p)
+ucs_status_t uct_ud_ep_create_connected(uct_iface_h iface_h,
+                                        const uct_device_addr_t *dev_addr,
+                                        const uct_iface_addr_t *iface_addr,
+                                        unsigned path_index,
+                                        size_t ep_dest_address_offset,
+                                        uct_ep_h *new_ep_p)
 {
+    uct_ud_iface_t *iface              = ucs_derived_of(iface_h,
+                                                        uct_ud_iface_t);
+    const uct_ib_address_t *ib_addr    = (const uct_ib_address_t*)dev_addr;
+    const uct_ud_iface_addr_t *if_addr = (const uct_ud_iface_addr_t*)iface_addr;
+    void *dest_address;
+    uct_ud_send_skb_t *skb;
     uct_ep_params_t params;
     ucs_status_t status;
     uct_ud_ep_t *ep;
     uct_ep_h new_ep_h;
 
+    uct_ud_enter(iface);
```

Comment:
why need here?

Reply:
it was in the UD TLs implementations:
- Verbs
https://github.com/openucx/ucx/blob/7f9a7f7a0d67a6143e163c378fbf76f47c15115f/src/uct/ib/ud/verbs/ud_verbs.c#L476
- MLX5
https://github.com/openucx/ucx/blob/7f9a7f7a0d67a6143e163c378fbf76f47c15115f/src/uct/ib/ud/accel/ud_mlx5.c#L602

So, not it is moved to to the base code